en/docs/architecture-diagram/architecture-diagram.md

```diff
@@ -0,0 +1,125 @@
+   }
+
+   remote function GetSupportedCurrencies(Empty value) returns GetSupportedCurrenciesResponse|error {
+      return {currency_codes: self.currencyMap.keys()};
+   }
+}
+```
+
+If other services interact with this service, this same identifier has to be provided upon creating the client endpoint. By doing so, it makes it possible to uniquely identify and link the service interactions between one another. 
+
+For example, the code snippet below calls on the `Currency Service` from a different component.
+
+```ballerina
+isolated function getSupportedCurrencies() returns string[]|error {
+   @display {
+      label: "CurrencyService",
+   id: "currency"
+   }
+   final CurrencyServiceClient currencyClient = check new ("http://localhost:9093");
+   GetSupportedCurrenciesResponse|grpc:Error supportedCurrencies = currencyClient->GetSupportedCurrencies({});
+   if supportedCurrencies is grpc:Error {
+      log:printError("failed to call getSupportedCurrencies from currency service");
+      return supportedCurrencies;
+   }
+   return supportedCurrencies.currency_codes;
+}
+```
+
+### Invoking service resources
+
+The resource invocations between services need to be done using [client access actions](https://ballerina.io/downloads/swan-lake-release-notes/swan-lake-2201.2.0#support-for-resource-methods-in-client-objects).
+
+## Explore the `Project Design View` features
+
+For demonstration purposes, this guide will use the [Google Cloud Platform demo](https://github.com/ballerina-guides/gcp-microservices-demo/tree/project-design-demo) to explain the different capabilities of the diagram. This sample has 10 services (9 GRPC and one HTTP) in 10 separate Ballerina packages. The `ui folder` is not a Ballerina package, hence would not make an impact.
+
+As discussed in the [Related Concepts](#a-project-workspace) section, each of these packages need to be loaded in to a VS Code workspace. This could be done using any of the following approaches.
+1. Use the VS Code workspace file that is available in the repository OR
+2. Add each Ballerina package to the workspace using the `Add folder to workspace` feature available VSCode
+
+If the workspace has been set up accurately, you would be able to see the following folder structure in the explorer. However, do note that the workspace name would be different if you set up the workspace from scratch.
+
+![Workspace Structure](images/workspace-structure.png)
+
+Once the workspace has been set up, the Ballerina design diagrams generator can be activated via the `Ballerina: Project Design` command. This command can be found in the VS Code command palette as seen below.
+![Extension Command](images/extension-command.gif)
+
+The execution of this command will open up a VS Code webview panel that will contain three types of design diagrams. The users can navigate between these diagrams using the side panel available on the diagram webview.
```

Comment:
```suggestion
The execution of this command will open up a VS Code web-view panel that will contain three types of design diagrams. You can navigate between these diagrams using the side panel available on the diagram web-view.
```

Reply:
Across VS Code and their documentation, they use `webview` : https://code.visualstudio.com/api/extension-guides/webview

Is it okay if we leave the same?